### PR TITLE
MULE-14311: Make ExtensionResourcesGeneratorAnnotationProcessor work …

### PR DESCRIPTION
**Pull request by estebanwasinger**
…in compile time

### CODE REVIEW

File: modules/extensions-spring-support/src/test/resources/models/metadata.json [hunk 1]
```diff
@@ -11978,7 +11977,8 @@
               "parent": "",
               "genericTypes": [],
               "isMap": false
-            }
+            },
+            "label": "Person"
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/test/resources/models/metadata.json]: I'm reviewing why the order changed

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java [hunk 2]
```diff
@@ -43,6 +47,15 @@
     this.loader = loader;
   }
 
+  protected ClassLoader getExtensionClassLoader() {
+    Class<?> extensionType = getExtensionType();
+    if (extensionType != null) {
+      return extensionType.getClassLoader();
+    } else {
+      return ExtensionModel.class.getClassLoader();
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java]: TCCL

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InjectedFieldsModelValidator.java [hunk 3]
```diff
@@ -102,7 +102,8 @@ public void visitObject(ObjectType objectType) {
                       validateType(model, type, DefaultEncoding.class);
                     }
                   } catch (Exception e) {
-                    problemsReporter.addWarning(new Problem(model, "Could not validate Class: " + e.getMessage()));
+                    //TODO - This is too verbose and fails always in AST Mode
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InjectedFieldsModelValidator.java]: ticket? no way to flag when we are in AST mode?

Comment (estebanwasinger) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InjectedFieldsModelValidator.java]: I will create a ticket

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java (hunk 2 above)
Comment (estebanwasinger) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java]: I Can't 😞 , If I do that I will get the Maven Plugin Classloader which doesn't have visibility to some Mule Classes.

Comment (alegmarra) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java]: Override this method on each implementation, and by default in the abstract return the TCC

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InjectedFieldsModelValidator.java [hunk 4]
```diff
@@ -61,113 +62,119 @@
...
+                @Override
+                public void visitObject(ObjectType objectType) {
+                  if (!objectType.getAnnotation(InfrastructureTypeAnnotation.class).isPresent()) {
+                    try {
+                      Class<?> type = getType(objectType, classLoaderModelProperty.getClassLoader());
+                      if (validatedTypes.add(type)) {
+                        validateType(model, type, DefaultEncoding.class);
+                      }
+                    } catch (Exception e) {
+                      //TODO - This is too verbose and fails always in AST Mode
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InjectedFieldsModelValidator.java]: 🤔  Still wrong

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java (hunk 2 above)
Comment (estebanwasinger) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractModelLoaderDelegate.java]: Implementations of this class are: `OperationModelLoaderDelegate`, `ParameterModelLoaderDelegate`, etc, which doesn't make sense that each one of this have to redefine these

**Pull request merged**